python/lsst/afw/geom/angle/angle.cc

```diff
@@ -0,0 +1,137 @@
+/*
+ * LSST Data Management System
+ * Copyright 2008-2016  AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include "pybind11/pybind11.h"
+
+#include "lsst/afw/geom/Angle.h"
+
+namespace py = pybind11;
+
+namespace lsst {
+namespace afw {
+namespace geom {
+namespace {
+
+using PyAngle = py::class_<Angle>;
+using PyAngleUnit = py::class_<AngleUnit>;
+
+template <typename OtherT>
+void declareAngleComparisonOperators(PyAngle & cls) {
+    cls.def("__eq__", [](Angle const& self, OtherT const& other) { return self == other; },
+            py::is_operator());
+    cls.def("__ne__", [](Angle const& self, OtherT const& other) { return self != other; },
+            py::is_operator());
+    cls.def("__le__", [](Angle const& self, OtherT const& other) { return self <= other; },
+            py::is_operator());
+    cls.def("__ge__", [](Angle const& self, OtherT const& other) { return self >= other; },
+            py::is_operator());
+    cls.def("__lt__", [](Angle const& self, OtherT const& other) { return self < other; }, py::is_operator());
+    cls.def("__gt__", [](Angle const& self, OtherT const& other) { return self > other; }, py::is_operator());
+}
+
+PYBIND11_PLUGIN(angle) {
+    py::module mod("angle");
+
+    /* AngleUnit */
+
+    PyAngleUnit clsAngleUnit(mod, "AngleUnit");
+
+    clsAngleUnit.def("__eq__", [](AngleUnit const& self, AngleUnit const& other) { return self == other; },
+                     py::is_operator());
+    clsAngleUnit.def("__ne__", [](AngleUnit const& self, AngleUnit const& other) { return !(self == other); },
+                     py::is_operator());
+    clsAngleUnit.def("_mul", [](AngleUnit const& self, double other) { return other * self; },
```

Comment:
I think this would be a good question to defer to [DM-9630](https://jira.lsstcorp.org/browse/DM-9630) -- `Angle` has enough inconsistencies in its operators even before this ticket.